client/components/content-containers/tcc-assessment-pool/edit/index.vue

```diff
@@ -9,68 +9,72 @@
     <v-alert :value="!hasAssessments" color="white" icon="mdi-information-variant">
       Click the button below to create first assessment.
     </v-alert>
-    <ul class="list-group">
+    <ul class="list-group pl-0">
       <assessment-item
         v-for="it in assessments"
         :key="it._cid"
         @selected="toggleSelect(it)"
         @save="saveAssessment"
-        @delete="requestRemoveConfirmation(it)"
+        @delete="$emit('deleteElement', it)"
         :assessment="it"
         :expanded="isSelected(it)" />
     </ul>
     <add-element
       @add="addAssessment"
       :include="['ASSESSMENT']"
-      :activity="activity"
+      :activity="container"
+      :position="nextPosition"
       :layout="false"
       large
       label="Add assessment" />
   </div>
 </template>
 
 <script>
-import { mapActions, mapGetters, mapMutations } from 'vuex';
 import AddElement from 'tce-core/AddElement';
-import AssessmentItem from './AssessmentItem';
-import capitalize from 'lodash/capitalize';
+import AssessmentItem from 'tce-core/AssessmentItem';
 import cuid from 'cuid';
-import EventBus from 'EventBus';
-import map from 'lodash/map';
-
-const appChannel = EventBus.channel('app');
+import filter from 'lodash/filter';
+import last from 'lodash/last';
+import sortBy from 'lodash/sortBy';
 
 export default {
-  name: 'assessments',
-  data() {
-    return {
-      selected: [],
-      allSelected: false
-    };
+  name: 'assessment-pool',
+  props: {
+    container: { type: Object, required: true },
+    elements: { type: Object, required: true }
   },
+  data: () => ({
+    selected: [],
+    allSelected: false
+  }),
   computed: {
-    ...mapGetters('editor', ['activity', 'assessments']),
-    hasAssessments() {
-      return this.assessments.length;
-    }
+    assessments() {
+      const activityId = this.container.id;
+      const assessments = filter(this.elements, { activityId });
+      return sortBy(assessments, 'position');
+    },
+    nextPosition() {
+      const lastItem = last(this.assessments);
+      return lastItem ? lastItem.position + 1 : 1;
+    },
+    hasAssessments: vm => vm.assessments.length
   },
   methods: {
-    ...mapActions('repository/tes', ['save', 'update', 'remove']),
-    ...mapMutations('repository/tes', ['add']),
     addAssessment(assessment) {
-      const data = { ...assessment, _cid: cuid() };
-      this.add(data);
-      this.selected.push(data._cid);
+      const cid = cuid();
+      this.$emit('addElement', { ...assessment, _cid: cid });
```

Comment:
I reverted the local state for unsaved assessments i introduced in this PR and moved it to the store where it was originally. The reason was the support for canceling. It needed to show a confirmation modal before removing the changes. With the store it is supported for every deletion event. With a local state I would need to do it inside the custom container or the `tce-question-container`. I believe the store is a better approach because it provides a singular interface for all custom containers and it does not introduce additional dependencies to components (example event bus). 

Looking at it now the usage of local state was a mistake. I believe the usage of local state for elements (maybe other things as well) in custom containers should be discouraged to keep the various implementation of custom containers streamlined and similar.